### PR TITLE
Precs to check if collection exists

### PR DESCRIPTION
**Pull request by maierlars**
Adding preconditions for jobs to check that the collection still exists.

### CODE REVIEW

**Comment by fceller** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/4897/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/4929/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/4932/

**Comment by maierlars** [comment]
Single unrelated failure in `testFillIndexFailure`.

File: arangod/Agency/FailedLeader.cpp [hunk 1]
```diff
@@ -115,13 +115,21 @@ void FailedLeader::rollback() {
     // Transactions
     payload = std::make_shared<Builder>();
     {
-      VPackObjectBuilder b(payload.get());
-      for (auto const c : cs) {
-        payload->add(planColPrefix + _database + "/" + c.collection +
-                         "/shards/" + c.shard,
-                     rb.slice());
+      VPackArrayBuilder a(payload.get());
```
Comment (neunhoef) [inline arangod/Agency/FailedLeader.cpp]: This does not look right. Earlier, the result was an object, now it is an array...

Comment (neunhoef) [reply arangod/Agency/FailedLeader.cpp]: OK, understood, `finish` distinguishes cases.

**Pull request merged**